Plugin Cubot/src/main/java/net/simon987/cubotplugin/Cubot.java

```diff
@@ -157,17 +173,26 @@ public char getMapInfo() {
      */
     @Override
     public void update() {
-
         if (currentAction == Action.WALKING) {
             if (spendEnergy(100)) {
                 if (!incrementLocation()) {
                     //Couldn't walk
                     currentAction = Action.IDLE;
+                }else{
+                    GameEvent event2 = new WalkDistanceEvent(this);
+                    GameServer.INSTANCE.getEventDispatcher().dispatch(event2);
                 }
             } else {
                 currentAction = Action.IDLE;
             }
         }
+        
+        int executionTime = timer.getTime();
+        if(executionTime > time && currentAction != Action.IDLE){
+            GameEvent event1 = new ExecutionTimeEvent(this, executionTime - time);
+            GameServer.INSTANCE.getEventDispatcher().dispatch(event1);
+        }
+        time = executionTime;
```

Comment:
Execution time is already calculated, (see comment below)

Reply:
removed